src/term.js

```diff
@@ -0,0 +1,218 @@
+'use strict'
+const Node = require('./node')
+
+class Term extends Node {
+  /**
+   * Assigns an index number and adds a BlankNode instance to the indices
+   * @internal
+   * @param bn The BlankNode instance to register
+   * @return {BlankNode} The updated BlankNode instance
+   */
+  static addBn(bn) {
+    if (bn.sI) {
+      throw new Error(`BlankNode ${bn} already registered`)
+    }
+
+    bn.sI = ++this.termIndex
+    this.termMap[bn.sI] = this.bnMap[bn.value] = bn
+
+    return bn
+  }
+
+  /**
+   * Assigns an index number and adds a Literal instance to the indices
+   * @internal
+   * @param lit The Literal instance to register
+   * @return {Literal} The updated Literal instance
+   */
+  static addLit(lit) {
+    if (lit.sI) {
+      throw new Error(`Literal ${lit} already registered`)
+    }
+
+    lit.sI = ++this.termIndex
+    const dtIndex = (lit.datatype || require('./xsd').string).sI
+    if (!this.litMap[dtIndex]) {
+      this.litMap[dtIndex] = []
+    }
+    if (lit.language) {
+      if (!this.litMap[dtIndex][lit.value]) {
+        this.litMap[dtIndex][lit.value] = []
+      }
+      this.termMap[lit.sI] = this.litMap[dtIndex][lit.value][lit.language] = lit
+    } else {
+      this.termMap[lit.sI] = this.litMap[dtIndex][lit.value] = lit
+    }
+
+    return lit
+  }
+
+  /**
+   * Assigns an index number and adds a NamedNode instance to the indices
+   * @internal
+   * @param nn {NamedNode} The NamedNode instance to register
+   * @param ln? {string} Property accessor-friendly string representation.
+   * @return {NamedNode} The updated NamedNode instance
+   */
+  static addNN(nn, ln) {
+    if (nn.sI) {
+      throw new Error(`NamedNode ${nn} already registered`)
+    }
+
+    nn.sI = ++this.termIndex
+    nn.term = ln
+    this.termMap[nn.sI] = this.nsMap[nn.value] = nn
```

Comment:
Checking whether `this.nsMap[nn.value]` already contains a value can aid in debugging, but introduces a runtime overhead. If rdflib has a 'dev build' it'd certainly be a good addition